### DISCUSSION TITLE
API Gateway. Quota usage retrieval

### DISCUSSION THREAD

**Discussion by jenriquesoriano** [INSPIRE-MIF/helpdesk-validator#977] [Announcements]
Dear Community,

taking into consideration the need raised by some of you, a method has been developed to retrieve the remaining quota associated with the API key, enabling the monitoring of the quota consumption when using the Validator API. 
This method has been implemented as part of the API Gateway itself and follows the following structure: 
[https://yzyiqfakm4.execute-api.eu-west-1.amazonaws.com/validator/v2/Usages/{API-Key}/](https://yzyiqfakm4.execute-api.eu-west-1.amazonaws.com/validator/v2/Usages/%7BAPI-key%7D/)
where API-key is the API key provided by the JRC for Validator API usage (please see #594).

By making a call to this endpoint, you will obtain the usage data for the last 10 days, returning the following information:

- date: date of usage
- uses: number of requests executed by that date
- remaining-uses: number of remaining requests to be consumed by that date

Please note that the quota is replenished on a weekly basis.

If you have any further questions, please do not hesitate to contact us.

Regards,

**Comment by vbombaerts** [comment]
Hello,
We tested the new functionality with an integration into a Geonetwork instance for validating metadata. It appears that the remaining quota is not updated directly when we launch a validation but only after a few hours. So in some cases we think we still have a number of request allowed while in reality the quota is already over. Could you please check this problem ?

Best regards,

Vincent

**Reply by jenriquesoriano** [reply]
Dear @vbombaerts 

we have conducted several tests to verify the update frequency of consumption data, from its usage to the update in the API. 
According to our results, this update occurs between 30 and 45 minutes after the API call is executed. 
In any case, if you obtain longer update times, we recommend taking the obtained update time into consideration when implementing automated processes using the validator's API.

Best regards,

**Comment by DanielMartinPerezdeLeon** [comment]
Dear Validator Community,

We’re pleased to inform you that the API Gateway has been removed (see [#1190](https://inspire.ec.europa.eu/validator/test-run/index.html?id=EID33f87631-885b-4955-aac8-f650fe13572c)). As a result, it is no longer necessary to retrieve an API key or manage quotas.

You can now access the Validator API directly and more easily. Here’s an example of how to run a test:


```
{
  "label": "Metadata Test",
  "executableTestSuiteIds": [
    "EID2be1480a-fe42-40b2-9420-eb0e69385c80",
    "EID0b86f7a3-2947-4841-823d-6a00d8e06d70",
    "EID7cceba68-e575-4429-9959-1b6b3d201b6d"
  ],
  "arguments": {
    "files_to_test": ".*",
    "tests_to_execute": ".*"
  },
  "testObject": {
    "id": "EID82ff0ea1-3120-4f6b-b633-012b06648208"
  }
}
```

If you have any questions or need assistance using the Validator, please don’t hesitate to contact us. We’re here to help.

Best regards,
The Validator Team

**Comment by fabiovinci** [comment]
As announced [here](https://github.com/INSPIRE-MIF/helpdesk-validator/issues/1190), the API Gateway has been discontinued, so this discussion is closed as it is no longer relevant.
